src/Microsoft.Identity.Client/Internal/OAuth2/OAuth2Client.cs

```diff
@@ -88,18 +88,38 @@ public void AddBodyParameter(string key, string value)
             }
 
             HttpResponse response = null;
-
-            if (method == HttpMethod.Post)
+            Uri endpointUri = CreateFullEndpointUri(endPoint);
+            var httpEvent = new HttpEvent()
+            {
+                HttpPath = endpointUri.AbsolutePath,
+                QueryParams = String.Join("&", MsalHelpers.ParseKeyValueList(endpointUri.Query, '&', false, true, requestContext).Keys)
+            };
+            Telemetry.GetInstance().StartEvent(requestContext.TelemetryRequestId, httpEvent);
+            try
             {
-                response = await
-                    HttpRequest.SendPost(CreateFullEndpointUri(endPoint), _headers, _bodyParameters,
-                        requestContext);
+                if (method == HttpMethod.Post)
+                {
+                    response = await
+                        HttpRequest.SendPost(CreateFullEndpointUri(endPoint), _headers, _bodyParameters,
+                            requestContext);
+                }
+                else
+                {
+                    response =
+                        await
+                            HttpRequest.SendGet(CreateFullEndpointUri(endPoint), _headers, requestContext);
+                }
+
+                httpEvent.HttpResponseStatus = (int) response.StatusCode;
+                httpEvent.UserAgent = response.UserAgent;
+                if (response.StatusCode != HttpStatusCode.OK)
+                {
+                    httpEvent.OauthErrorCode = JsonHelper.DeserializeFromJson<TokenResponse>(response.Body).Error;
+                }
             }
-            else
+            finally
             {
-                response =
-                    await
-                        HttpRequest.SendGet(CreateFullEndpointUri(endPoint), _headers, requestContext);
+                Telemetry.GetInstance().StopEvent(requestContext.TelemetryRequestId, httpEvent);
```

Comment:
I think we can consider 113-118 to be noise overhead and try not to optimize for it.